### ISSUE TITLE
User interface ideas

### ISSUE THREAD

**Issue by jackfirth** [jackfirth/resyntax#27]
How should `resyntax` surface suggestions to users? I can think of several possibilities:

- A DrRacket plugin that shows you suggestions for the current file. Maybe something similar for `racket-mode`, maybe even something hooked up to [racket-langserver](https://github.com/jeapostrophe/racket-langserver).
- A GitHub integration that makes suggestions at code review time, possibly with some way to automatically commit them to the pull request branch.
- A GitHub Actions workflow that automatically opens a pull request refactoring your code once a day/week/etc.
- A command-line tool for refactoring files directly.
- A web app where you can tell a bot to refactor a racket package and open a pull request.

Ideas and opinions welcomed!

**Comment by jackfirth** [comment]
Gave up on the github pull request integration approach for now. Seems there isn't yet a way for github actions to produce annotations: https://github.community/t/what-are-annotations/16173

**Comment by jackfirth** [comment]
New user interface idea: a Discord bot (and/or a Slack bot) so that people can run `resyntax` over code in chatrooms and get suggestions that way. Would be especially useful in the `#help` channel of the Racket Discord server, where students often show up and share code looking for advice.

**Comment by dannypsnl** [comment]
Do you have plan to make some functions as interface? Then racket-mode or racket-langserver can reuse them.

**Comment by jackfirth** [comment]
Yes, though that'd be the programmatic interface. I'm not sure what it will look like yet.

**Comment by dannypsnl** [comment]
Maybe a function type is `file -> (listof suggestion)`, a `suggestion` would be a `text -> text`? From my understanding this model can fit racket-langserver and racket-mode both.

**Comment by jackfirth** [comment]
There's some details that make that difficult to match up with Resyntax's internal models. But more generally, I would much rather Resyntax itself own the integration with the langserver, so there's no need for the interface used there to be something public that has to be kept stable and committed to for all other tools to use.
